machine_spec.py

```diff
@@ -270,6 +270,8 @@ def detect_arch() -> str:
 
         "mips64":     "mips64r2",
         "mips64el":   "mips64r2",
+
+        "powerpc":    "ppc",
```

Comment:
This mapping seems incorrect and shouldn't be needed, since Meson's CPU family value is `ppc` according to their [reference table](https://mesonbuild.com/Reference-tables.html#cpu-families).